practice/caesar/site.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+/**
+ * One function converts a string into a ceaser cipher based on a key,
+ * the other returns a translated string based on a decipher key.
+ *
+ * Cipher example: 'cats', 1 -> 'dbut'
+ * Decipher example:  fqiu', 2 -> 'dogs'
+ */
+
+
+var ALPHA_TO_NUMBER = 'abcdefghijklmnopqrstuvwxyz'.split('');
+/** Takes in a word and a key, and returns a caeser cypher version of
+ * the word, if the word has any punctuation in it, the punctuation remains as
+ * it was.
+ *
+ * It does this by assigning each letter an alphanumeric index, then
+ * increasing that alphanumeric index by the value of the key, wrapping around
+ * the alpbanumeric index. */
+function encryptWord(plainString, key) {
+  var lowersString = _.toLower(plainString);
+  var workingLetters = lowersString.split('');
+  var wordCipher = _.reduce(
+    workingLetters,
+    function(runningWord, letter) {
+      if (ALPHA_TO_NUMBER.indexOf(letter) >= 0) {
+        var oldAlphaIndex = ALPHA_TO_NUMBER.indexOf(letter);
+        if (key + oldAlphaIndex >= ALPHA_TO_NUMBER.length) {
+          var newAlphaIndex = key + oldAlphaIndex - ALPHA_TO_NUMBER.length;
+        } else {
+          var newAlphaIndex = oldAlphaIndex + key;
+        } return runningWord + ALPHA_TO_NUMBER[newAlphaIndex];
+      } else {
+        return runningWord + letter;
+      }},
+    '');
```

Comment:
I think this is less of a reduction and more of a mapping and a string join. The "item level" problem is rotating a single letter, and there's no interaction between the letters. If there's no interaction, then a mapping is a much cleaner way to solve the problem since it separates "item" from "list" level better.